### DISCUSSION TITLE
Are permutation prompts supported?

### DISCUSSION THREAD

**Discussion by adumont** [lllyasviel/Fooocus#1162] [Q&A]
Are permutation prompts supported in Fooocus?

Example: `a {red, green, yellow} bird`

Reference: [Midjourney Permutation Prompts](https://docs.midjourney.com/docs/permutations)

**Comment by rubenmejiac** [comment]
Indeed, I think it would be very useful to have the same or similar syntax for dynamic prompts as ComfyUI, i.e.:

`a {red|green|yellow} bird`
`scene of a landscape in {0.5::summer|0.1::autumn|0.3::winter|0.1::spring}`
`{2$$chocolate|vanilla|strawberry} ice cream`

Also for wildcard uses:
`{ice cream flavours {2$$__flavours__}`
```
# seasons.txt
summer
{autumn|fall}
winter
spring
```

You can find the ComfyUI dynamic prompts reference here:
[https://github.com/adieyal/sd-dynamic-prompts/blob/main/docs/SYNTAX.md#variants](https://github.com/adieyal/sd-dynamic-prompts/blob/main/docs/SYNTAX.md#variants)

**Reply by aolko** [reply]
wildcards are supported
